### PR TITLE
[codex] run AutoResearch candidates with proposed config

### PR DESCRIPTION
**Pull request by RaychevAngel**
## Summary
- seed `configs/current.json` from the active AutoResearch state during init so proposal patching starts from the actual baseline config
- submit pending proposal patches to the Tinker runner during iteration runs, instead of rerunning the unpatched baseline hyperparameters
- add regression tests for direct `run_node`, `propose_node` -> `run_node`, and init config seeding

## Why
The full AutoResearch graph looked wired after the Tinker integration, but the proposal path had a silent failure mode: `propose_node` patched `configs/current.json`, while `run_node` built `TrainingConfig` only from `state["current_config"]`. That meant candidate experiments could ignore the proposed hyperparameter change and retrain the baseline config.

## Validation
- `python3 -m compileall src`
- `uvx --with pytest --with anthropic --with langgraph --with tinker --with tinker-cookbook python -m pytest tests/test_tinker_runner.py -q` -> 16 passed
- `uvx --with pytest --with anthropic --with langgraph --with tinker --with tinker-cookbook python -m pytest tests/test_autoresearch.py tests/test_cost_manager.py tests/test_tinker_runner.py -q` -> 50 passed
- expanded stacked merge smoke (#39 + #45 + #43 + #40 + #42 + #44 + #46): broad non-live suite -> 186 passed, 4 skipped
- live candidate-config graph smoke on expanded stack: offline synthetic DataGen -> curation -> DecisionEngine -> baseline Tinker + one deterministic proposal, both 2 steps; captured learning rates were `0.0001` then `0.0002`; Tinker reported `$0.000284`
- `git diff --check`

### CODE REVIEW

**Comment by RaychevAngel** [comment]
Superseded by consolidated replacement PR(s): #116. This old draft was part of the fragmented AutoResearch/budget stack.
